File: AtlasAI/WebhookIntegration/src/utils/logger.ts
type LogLevel = 'info' | 'warn' | 'error' | 'debug';

function log(level: LogLevel, message: string, meta?: Record<string, unknown>): void {
  const timestamp = new Date().toISOString();
  const entry = meta
    ? `[${timestamp}] [${level.toUpperCase()}] ${message} ${JSON.stringify(meta)}`
    : `[${timestamp}] [${level.toUpperCase()}] ${message}`;

  if (level === 'error') {
    console.error(entry);
  } else {
    console.log(entry);
  }
}

export const logger = {
  info: (message: string, meta?: Record<string, unknown>) => log('info', message, meta),
  warn: (message: string, meta?: Record<string, unknown>) => log('warn', message, meta),
  error: (message: string, meta?: Record<string, unknown>) => log('error', message, meta),
  debug: (message: string, meta?: Record<string, unknown>) => log('debug', message, meta),
};

File: Atlas/Engine/Core/Logger.cpp
#include "Logger.h"
#include <iostream>

namespace atlas {

void Logger::Init() {
    Info("Logger initialized");
}

void Logger::Info(const std::string& msg) {
    std::cout << "[INFO] " << msg << std::endl;
}

void Logger::Warn(const std::string& msg) {
    std::cout << "[WARN] " << msg << std::endl;
}

void Logger::Error(const std::string& msg) {
    std::cerr << "[ERROR] " << msg << std::endl;
}

}
